src/lib.rs

```diff
@@ -16,23 +17,22 @@ const ESCAPE_KEY: &str = "Escape";
 
 const STORAGE_KEY: &str = "todos-seed";
 
-// ------ Url path parts ------
-const ACTIVE: &str = "active";
-const COMPLETED: &str = "completed";
-
+add_router!();
 // ------ ------
 //     Init
 // ------ ------
 
 fn init(url: Url, orders: &mut impl Orders<Msg>) -> Model {
-    orders.subscribe(Msg::UrlChanged);
+    router()
+        .init(url)
+        .set_handler(orders, move |subs::UrlRequested(requested_url, _)| {
+            router().confirm_navigation(requested_url)
+        });
```

Comment:
Why we need to define the handler?

Reply:
It is what we talked about before.

Moving `confirm_navigation` from `update()` to `init()`

Reply:
Well. I am not sure what I could do there except maybe improve the name of the method

Reply:
We also had the issue with life time and stuff. And using it this way solved all the issues we had.

Reply:
Hey, I will look at this tomorrow. I have been pretty unavailable for the last few days. Hopefully I will come with a solution or better suggestion soon.

Reply:
Well, just for testing I changed to

```rust
  router()
        .init(url)
        .subscribe(orders.subscribe_with_handle(
        |subs::UrlRequested(requested_url, _)| router().confirm_navigation(requested_url),
    ));
```

It is more code, but it is more transparent as well with nothing "under the hood". I am not sure what is best actually.

Reply:
`subscribe` looks better than `set_handler` imho